src/contact.rs

```diff
@@ -34,34 +30,60 @@ use crate::{chat, stock_str};
 ///
 /// Some contact IDs are reserved to identify special contacts.  This
 /// type can represent both the special as well as normal contacts.
-#[derive(
-    Debug, Copy, Clone, Default, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Deserialize,
-)]
+#[derive(Debug, Copy, Clone, Default, PartialEq, Eq, Hash, Serialize, Deserialize)]
```

Comment:
by pr description, i assume the removal from `PartialOrd, Ord` is on purpose, but i do not really understand why :)

was it just needed because of the `BTreeMap` decision below?

Reply:
Rather the other way around, the `BTreeMap` change was needed because I removed `Ord`.  The reason to remove ordering was to make this identifier more opaque to rust code, some part of my brain thinks it's more logical if an identifier does not need to be ordered.  (Beside, our other "id newtypes" also have `.is_special()`)

At least in theory a small `BTreeSet` for a number should be faster than a `HashSet`, hence the downside of this.

Reply:
thanks for the more detailed explanations!